addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -356,38 +275,130 @@ def _compute_from_employee_id(self):
             allocation.number_of_days = allocation.number_of_days_display
             if allocation.type_request_unit == 'hour':
                 allocation.number_of_days = allocation.number_of_hours_display / (allocation.employee_id.sudo().resource_calendar_id.hours_per_day or HOURS_PER_DAY)
+            if allocation.accrual_plan_id.time_off_type_id != allocation.holiday_status_id:
+                allocation.accrual_plan_id = False
+            if allocation.allocation_type == 'accrual' and not allocation.accrual_plan_id and allocation.holiday_status_id:
+                allocation.accrual_plan_id = self.env['hr.leave.accrual.plan'].search([('time_off_type_id', '=', allocation.holiday_status_id.id)], limit=1)
 
-            # set default values
-            if not allocation.interval_number and not allocation._origin.interval_number:
-                allocation.interval_number = 1
-            if not allocation.number_per_interval and not allocation._origin.number_per_interval:
-                allocation.number_per_interval = 1
-            if not allocation.unit_per_interval and not allocation._origin.unit_per_interval:
-                allocation.unit_per_interval = 'hours'
-            if not allocation.interval_unit and not allocation._origin.interval_unit:
-                allocation.interval_unit = 'weeks'
-
-            if allocation.holiday_status_id.validity_stop and allocation.date_to:
-                new_date_to = datetime.combine(allocation.holiday_status_id.validity_stop, time.max)
-                if new_date_to < allocation.date_to:
-                    allocation.date_to = new_date_to
-
-            if allocation.allocation_type == 'accrual':
-                if allocation.holiday_status_id.request_unit == 'hour':
-                    allocation.unit_per_interval = 'hours'
-                else:
-                    allocation.unit_per_interval = 'days'
+    @api.depends('accrual_plan_id', 'employee_id')
+    def _compute_accrual_level_id(self):
+        """
+        The current accrual_id depends on conditions set on the accrual and the employee start date.
+        This methods get all accrual potential values and chose the best suited:
+           * The start_date (according to employee is past
+           *
+        """
+        results = {}
+        for allocation in self.filtered(lambda i: i.accrual_plan_id and i.employee_id):
+            start_date = allocation.create_date
+            first_day = datetime.combine(start_date, time(0, 0, 0))
+            accruals_dates = allocation.accrual_plan_id.level_ids._get_accrual_values(first_day)
+            # The selection of the accrual_id (the current step) depends on
+            # the seniority of the employee: it must be sufficient.
+            canditate_cur = list(filter(lambda a: a['sufficient_seniority'] is True, accruals_dates))
+            if canditate_cur:
+                current_step = max(canditate_cur, key=lambda x: x['start_after'])
```

Comment:
Warning, this is broken. If you create two steps: 
1) every week, start_count = 1
2) every month, start_month = 1
The selected current step is not the good one. because start_after is the start_count value.
I would add a value: seniority in accruals_dates result and do that:
current_step = max(canditate_cur, key=lambda x: x['seniority'])
What do you think?